li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -356,19 +358,27 @@ public void unsubscribe() {
         }
       }
 
-      _lastProcessedResult = _consumerRecordsProcessor.process(rawRecords);
-      processedRecords = _lastProcessedResult.consumerRecords();
-      // Clear the internal reference.
-      _lastProcessedResult.clearRecords();
-      // Rewind offset if there are processing exceptions.
-      seekToCurrentOffsetsOnRecordProcessingExceptions();
-
-      // this is an optimization
-      // if no records were processed try to throw exception in current poll()
-      if (processedRecords.isEmpty()) {
-        crpe = handleRecordProcessingException(null);
-        if (crpe != null) {
-          throw crpe;
+      /* Hack for passthrough mirroring: since Brooklin is the only user of passthrough feature and they only
+       * use raw <byte[], byte[]> consumer, there's no need to perform large message processing in this case,
+       * and should just return the consumed records directly to the caller.
+       */
+      if (_passthroughEnabled) {
+        processedRecords = (ConsumerRecords<K, V>) rawRecords;
```

Comment:
if they are using `<byte[], byte[]>` consumer, is there a need for casting here?

Reply:
yes in their use case there's no need for casting, but the function signature here returns a typed ConsumerRecords<K, V>, thus the cast.